src/Bicep.RegistryModuleTool/ModuleFiles/MainArmTemplateFile.cs

```diff
@@ -98,36 +118,43 @@ public MainArmTemplateFile WriteToFileSystem(IFileSystem fileSystem)
             return this;
         }
 
-        private static MainArmTemplateParameter ToParameter(JsonProperty parameterProperty)
+        private MainArmTemplateParameter ToParameter(JsonProperty parameterProperty)
         {
-            string name = parameterProperty.Name;
-            string type = parameterProperty.Value.GetProperty("type").ToNonNullString();
-            bool required = !parameterProperty.Value.TryGetProperty("defaultValue", out _);
-            string? description = TryGetDescription(parameterProperty.Value);
+            var parameters = this.armTemplate.Parameters;
 
-            return new(name, type, required, description);
+            string name = parameters[parameterProperty.Name].Name;
+            string type = GetPrimitiveTypeName(parameters[parameterProperty.Name].TypeReference);
+            bool isRequired = parameters[parameterProperty.Name].IsRequired;
+            string? description = parameters[parameterProperty.Name].Description;
+
+            return new(name, type, isRequired, description);
         }
 
-        private static MainArmTemplateOutput ToOutput(JsonProperty outputProperty)
+        private MainArmTemplateOutput ToOutput(JsonProperty outputProperty)
         {
-            string name = outputProperty.Name;
-            string type = outputProperty.Value.GetProperty("type").ToNonNullString();
-            string? description = TryGetDescription(outputProperty.Value);
 
-            return new(name, type, description);
-        }
+            var outputs = this.armTemplate.Outputs.ToImmutableDictionaryExcludingNullValues(x => x.Name, new MyEqualityComparer());
```

Comment:
```suggestion
var outputs = this.armTemplate.Outputs.ToImmutableDictionaryExcludingNullValues(x => x.Name, StringComparer.Ordinal.);
```

`MyEqualityComparer` below can be removed.